browser/brave_wallet/brave_wallet_policy_browsertest.cc

```diff
@@ -0,0 +1,165 @@
+/* Copyright (c) 2022 The Brave Authors. All rights reserved.
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#include "base/strings/stringprintf.h"
+#include "brave/app/brave_command_ids.h"
+#include "brave/browser/brave_wallet/brave_wallet_context_utils.h"
+#include "brave/browser/brave_wallet/brave_wallet_service_factory.h"
+#include "brave/browser/ui/brave_browser.h"
+#include "brave/browser/ui/sidebar/sidebar_controller.h"
+#include "brave/browser/ui/sidebar/sidebar_model.h"
+#include "brave/browser/ui/sidebar/sidebar_service_factory.h"
+#include "brave/components/brave_wallet/common/common_util.h"
+#include "brave/components/brave_wallet/common/pref_names.h"
+#include "brave/components/sidebar/sidebar_item.h"
+#include "brave/components/sidebar/sidebar_service.h"
+#include "chrome/browser/ui/browser.h"
+#include "chrome/browser/ui/browser_command_controller.h"
+#include "chrome/browser/ui/tabs/tab_strip_model.h"
+#include "chrome/browser/ui/views/frame/browser_view.h"
+#include "chrome/test/base/in_process_browser_test.h"
+#include "chrome/test/base/ui_test_utils.h"
+#include "components/policy/core/browser/browser_policy_connector.h"
+#include "components/policy/core/common/mock_configuration_policy_provider.h"
+#include "components/policy/core/common/policy_map.h"
+#include "components/policy/policy_constants.h"
+#include "components/prefs/pref_service.h"
+#include "components/user_prefs/user_prefs.h"
+#include "content/public/browser/browser_context.h"
+#include "content/public/browser/web_contents.h"
+#include "content/public/test/browser_test.h"
+#include "content/public/test/browser_test_utils.h"
+#include "content/public/test/navigation_handle_observer.h"
+#include "extensions/common/constants.h"
+#include "url/gurl.h"
+
+namespace policy {
+
+class BraveWalletPolicyTest : public InProcessBrowserTest,
+                              public ::testing::WithParamInterface<bool> {
+ public:
+  BraveWalletPolicyTest() = default;
+  ~BraveWalletPolicyTest() override = default;
+
+  void SetUpInProcessBrowserTestFixture() override {
+    EXPECT_CALL(provider_, IsInitializationComplete(testing::_))
+        .WillRepeatedly(testing::Return(true));
+    BrowserPolicyConnector::SetPolicyProviderForTesting(&provider_);
+    PolicyMap policies;
+    policies.Set(key::kBraveWalletDisabled, POLICY_LEVEL_MANDATORY,
+                 POLICY_SCOPE_USER, POLICY_SOURCE_PLATFORM,
+                 base::Value(IsBraveWalletDisabledTest()), nullptr);
+    provider_.UpdateChromePolicy(policies);
+  }
+
+  void PreRunTestOnMainThread() override {
+    InProcessBrowserTest::PreRunTestOnMainThread();
+
+    auto* service = sidebar::SidebarServiceFactory::GetForProfile(profile());
+    // Enable sidebar explicitly because sidebar option is different based on
```

Comment:
nit: sidebar show option state doesn't affect this test. It just controls visibilty.
Service will initialize its data always.

Reply:
👍 Removed in 1492771abf3fe4acc1e0cd44f8ad564292873ff3